File: LeetCode-Java/src/Link/leetcode/easy/LeetCode21.java
package Link.leetcode.easy;

public class LeetCode21 {
    public ListNode mergeTwoLists(ListNode l1, ListNode l2) {
        ListNode dummyNode = new ListNode(-1);
        ListNode indexNode = dummyNode;
        while (l1 != null && l2 != null) {
            if (l1.val < l2.val) {
                indexNode.next = l1;
                l1 = l1.next;
            } else {
                indexNode.next = l2;
                l2 = l2.next;
            }
            indexNode = indexNode.next;
        }
        if (l1 != null) {
            indexNode.next = l1;
        } else if (l2 != null){
            indexNode.next = l2;
        }
        return dummyNode.next;
    }
    public static void main(String[] args) {
        LeetCode21 leetCode21 = new LeetCode21();
        ListNode l1 = ListNode.createWithNums(new int[]{1, 2, 4});
        ListNode l2 = ListNode.createWithNums(new int[]{1, 3, 4});
        leetCode21.mergeTwoLists(l1, l2);
    }
}

File: LeetCode-Swift/LeetCode-Swift/Base Data Struct/Link/LeetCode21.swift
//
//  LeetCode21.swift
//  LeetCode-Swift
//
//  Created by Liang on 2020/5/1.
//  Copyright © 2020 imieutan. All rights reserved.
//

import Foundation
/**
 21. 合并两个有序链表

 */
public class LeetCode21 {
    func mergeTwoLists(_ l1: ListNode?, _ l2: ListNode?) -> ListNode? {
        var list1 = l1, list2 = l2
        let dummyNode = ListNode(-1)
        var indexNode = dummyNode
        while list1 != nil || list2 != nil {
            if list2 == nil {
                indexNode.next = list1
                break
            }
            if list1 == nil {
                indexNode.next = list2
                break
            }
            if list1!.val > list2!.val {
                indexNode.next = list2
                list2 = list2!.next
            } else {
                indexNode.next = list1
                list1 = list1!.next
            }
            indexNode = indexNode.next!
        }
        return dummyNode.next
    }
}

